code/modules/antagonists/morph/morph.dm

```diff
@@ -213,7 +213,10 @@
 	visible_message("<span class='warning'>[src] suddenly twists and changes shape, becoming a copy of [target]!</span>", \
 					"<span class='notice'>You twist your body and assume the form of [target].</span>")
 	appearance = target.appearance
-	copy_overlays(target)
+	copy_overlays(target, TRUE)
+	if(length(target.vis_contents))
+		for(var/atom/A as() in target.vis_contents)
+			add_overlay(A)
```

Comment:
```suggestion
	if(length(target.vis_contents))
		add_overlay(target.vis_contents)
```

Internal BYOND magic: copying `appearance` also copies overlays (the reason why adding atom as an overlay also adds that atom's overlays)
Original code: hey let's copy overlays separately
Me: Surprised pikachu picture

Anyways, this should be the cleanest way to achieve this.

Reply:
Hmm i am sure i already tried something like that but yeah you are right it works can't argue with the result.